src/main/java/com/github/cameltooling/lsp/internal/completion/CompletionResolverUtils.java

```diff
@@ -16,18 +16,31 @@
  */
 package com.github.cameltooling.lsp.internal.completion;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.camel.parser.helper.CamelXmlHelper;
 import org.eclipse.lsp4j.CompletionItem;
 import org.eclipse.lsp4j.Position;
 import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextDocumentItem;
 import org.eclipse.lsp4j.TextEdit;
+import org.w3c.dom.Node;
 
+import com.github.cameltooling.lsp.internal.instancemodel.CamelURIInstance;
 import com.github.cameltooling.lsp.internal.instancemodel.CamelUriElementInstance;
+import com.github.cameltooling.lsp.internal.instancemodel.PathParamURIInstance;
+import com.github.cameltooling.lsp.internal.parser.ParserXMLFileHelper;
 
 /**
  * @author lheinema
  */
 public class CompletionResolverUtils {
 	
+	private static final List<String> POSSIBLE_DIRECT_REFERENCE = Arrays.asList("direct","direct-vm", "seda", "vm", "amqp", "jms", "activemq", "broker");
```

Comment:
for JMS I'm not sure that itis the correct behavior. The "name" to reference might not be the second element.

for instance 
![image](https://user-images.githubusercontent.com/1105127/46077321-bde2aa00-c190-11e8-8c07-03fa231c9b41.png)
I think that the completion for jms:another is wrong because "another" is a topic although when not specified it is a queue